src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPEditorImprovementsBackspaceHandler.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.formatting;
+
+import com.intellij.codeInsight.editorActions.BackspaceHandlerDelegate;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LSPIJEditorUtils;
+import com.redhat.devtools.lsp4ij.client.features.LSPEditorBehaviorFeature;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Improves handling of backspace for the escaped opening character of a paired set, specifically quote characters and
+ * brace pairs.
+ */
+public class LSPEditorImprovementsBackspaceHandler extends BackspaceHandlerDelegate {
+
+    @Override
+    public void beforeCharDeleted(char charDeleted,
+                                  @NotNull PsiFile file,
+                                  @NotNull Editor editor) {
+        // No-op
+    }
+
+    @Override
+    public boolean charDeleted(
+            char charDeleted,
+            @NotNull PsiFile file,
+            @NotNull Editor editor) {
+        if (LSPIJEditorUtils.isSupportedAbstractFileTypeOrTextMateFile(file) &&
+            LSPEditorBehaviorFeature.enableStringLiteralImprovements(file) &&
```

Comment:
Immediate feature flag check so if this feature is disabled in client config, this is a no-op.